apiserver/facades/client/sshclient/facade.go

```diff
@@ -228,3 +249,89 @@ func (facade *Facade) Leader(entity params.Entity) (params.StringResult, error)
 	}
 	return result, nil
 }
+
+// ModelCredentialForSSH did not exist prior to v4.
+func (*FacadeV3) ModelCredentialForSSH(_, _ struct{}) {}
+
+// ModelCredentialForSSH returns a cloud spec for ssh purpuse.
+// This facade is only used for k8s model.
+func (facade *Facade) ModelCredentialForSSH() (params.CloudSpecResult, error) {
+	var result params.CloudSpecResult
+
+	canModifyModel, err := facade.authorizer.HasPermission(permission.AdminAccess, facade.backend.ModelTag())
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	isSuperUser, err := facade.authorizer.HasPermission(permission.SuperuserAccess, facade.backend.ControllerTag())
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	if !canModifyModel && !isSuperUser {
+		return result, apiservererrors.ErrPerm
+	}
+
+	model, err := facade.backend.Model()
+	if err != nil {
+		result.Error = apiservererrors.ServerError(err)
+		return result, nil
+	}
+	if model.Type() != state.ModelTypeCAAS {
+		result.Error = apiservererrors.ServerError(errors.NotImplementedf("facade ModelCredentialForSSH for non %q model", state.ModelTypeCAAS))
+		return result, nil
+	}
+
+	spec, err := facade.backend.CloudSpec()
+	if err != nil {
+		result.Error = apiservererrors.ServerError(err)
+		return result, nil
+	}
+	if spec.Credential == nil {
+		result.Error = apiservererrors.ServerError(errors.NotValidf("cloud spec %q has empty credential", spec.Name))
+		return result, nil
+	}
+
+	token, err := facade.getExecSecretToken(spec, model)
+	if err != nil {
+		result.Error = apiservererrors.ServerError(err)
+		return result, nil
+	}
```

Comment:
This should be an internal detail of the k8s provider.
We should add a method to return a credential for a specified secret name and call that. This method and the lines below can all be implemented in that k8s provider api call which takes as a parameter the seret name, ie in this case k8sprovider.ExecRBACResourceName

Reply:
As discussed, we keep the current implementation because the GetSecretToken can be reused for other purposes like secret policy token (and they just need the token but not the `credential`.).